### PR TITLE
BE-SwimMeet model, serializer and view

### PR DESCRIPTION
**Pull request by MarisiaS**
This PR addresses issue #27  

**Implementation**

1. **backend/api/models.py**
Implemented the SwimMeet model with the following fields:

- name: required
- date: required
- time: Can be null
- site: This is a foreignKey to Site. Set on_delete to cascate because without a site configurate a swim meet is not posible (num lanes is needed to generate the heats)

2. **backend/api/migrations/0011_swimmeet.py**
After creating the model, I ran the `makemigrations` command and the migration for the SwimMeet model was created.

3.  **backend/api/serializers/SwimMeetSerializer.py**
Created a Model serializer for the SwimMeet model. Included all the fields plus the site name.

4. **backend/api/views/EventTypeView.py**
Created a basic Model View  set for SwimMeet.
The data is order by date. 
Search by name is allow

5. **backend/api/urls.py**
Added a router to generate the endpoints for SwimMeet.

6. **backend/api/fixtures/swim_meet.json**

Added 2 swim meets

**Swagger**

The endpoints are accessible through Swagger.

I tried each of the new endpoints successfully. I was able to create, delete, patch and get an swim meet.
When posting and patching, I got the expected error for entering invalid data.

### CODE REVIEW

File: backend/api/models.py [hunk 1]
```diff
@@ -110,3 +110,11 @@ class Session(models.Model):
     time = models.TimeField()
     coach = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name='coach_group')
     school = models.ForeignKey(School, on_delete=models.CASCADE, related_name='school_group')
+
+class SwimMeet(models.Model):
+    name = models.CharField(max_length=255, null=False, blank=False)
+    date = models.DateField(null=True, blank=True)
```
Comment (viriponce) [inline backend/api/models.py:116]: The user story and the description indicate that this field is required therefore null and blank attributes should be false.

File: backend/api/views/SwimMeetView.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from rest_framework.permissions import IsAuthenticated
+from api.models import SwimMeet
+from api.serializers.SwimMeetSerializer import SwimMeetSerializer
+from drf_spectacular.utils import extend_schema
+import logging
+
+logger = logging.getLogger('django')
+
+
+@extend_schema(tags=['SwimMeet'])
```
Comment (viriponce) [inline backend/api/views/SwimMeetView.py:12]: is a space missing between words?

**Review by viriponce — CHANGES_REQUESTED** [review]
The endpoints are working as expected, I was able to post, patch, delete and get swim meets.

Just a couple of observations:
Issue #27 calls for fixtures, those are not included in this PR.
I see you added search and sort functionality, should we add then the insensitive collation in the model?

**Comment by MarisiaS** [comment]
> Just a couple of observations: Issue #27 calls for fixtures, those are not included in this PR. I see you added search and sort functionality, should we add then the insensitive collation in the model?

I just added sorting by date, so the insensitive collation is not needed.

**Review by viriponce — APPROVED** [review]
LGTM
Thank you Marisia!

**Comment by viriponce** [comment]
> > Just a couple of observations: Issue #27 calls for fixtures, those are not included in this PR. I see you added search and sort functionality, should we add then the insensitive collation in the model?
> 
> I just added sorting by date, so the insensitive collation is not needed.

Sorry, I got confuse with name but name is a search filter. :)

**Pull request merged**